cmd/kube-apiserver/app/server.go

```diff
@@ -374,7 +375,7 @@ func Run(s *options.APIServer) error {
 			ReadWritePort:             s.SecurePort,
 			PublicAddress:             s.AdvertiseAddress,
 			Authenticator:             authenticator,
-			SupportsBasicAuth:         len(s.BasicAuthFile) > 0,
+			SupportsBasicAuth:         len(s.BasicAuthFile) > 0 || s.PamPassword,
```

Comment:
This flag controls sending basic auth challenges, which will prompt browsers for  passwords, without any way to log out, and without any cross site request forgery protection. We should not enable this flag for more password authentication methods without addressing those issues